force_bdss/io/workflow_reader.py

```diff
@@ -80,36 +64,21 @@ def read(self, path):
 
         Parameters
         ----------
-        file: File
-            A file object containing the data of the workflow in the
-            appropriate json format.
+        path: str
+            A path to file containing the serialized data of the workflow.
 
         Returns
         -------
         Workflow
             An instance of the model tree, rooted at Workflow.
-
-        Raises
-        ------
-        InvalidFileException
-            Raised if the file is corrupted or cannot be read by this reader.
-
-        InvalidVersionException
-            Raised if the version is not supported.
-
-        MissingPluginException
-            The file cannot be opened because it needs a plugin that is not
-            available.
-
-        ModelInstantiationFailedException
-            When instantiating the model for a given plugin, an exception
-            occurred. This is likely due to a coding error in the plugin.
-
         """
         json_data = self.load_data(path)
 
-        _ = self._extract_version(json_data)
-        workflow = self._extract_workflow(json_data)
+        self.workflow_format_version = self._extract_version(json_data)
+
+        workflow_data = self._preprocess_workflow_data(json_data)
+
+        workflow = Workflow.from_json(self.factory_registry, workflow_data)
```

Comment:
Is it possible to create a `parse_data` pubic method to expose this section without needing to read the file?

It would be handy to use this in `force_wfmanager` (and would also make the mocking / unit testing a lot simpler!), where we can load in project files without having to read the same file twice.

Reply:
I quite agree, that would be very useful. I am working on this now. Thank you!

Reply:
Thank you, now we can use `WorkflowReader.parse_data(json_data)`.